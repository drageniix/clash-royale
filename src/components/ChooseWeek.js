import React from 'react'
import { setNewWeek } from '../redux/actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

export class ChooseWeek extends React.Component {
    setNewWeek = index => () => {
        this.props.setNewWeek(this.props.lastWeeks, index)
    }

    render(){
        return (
            <span className="weekSelection">
                <a>History</a>
                <div className="weekSelection__weeks">
                    {this.props.lastWeeks.map((item, index) => (
                        <li className={(this.props.current === index ? 'selectedWeek' : '')} 
                            key={index} 
                            onClick={this.setNewWeek(index)}>
                            {item.display}</li>
                    ))}
                </div>
            </span>
        )
    }

} 

const mapStateToProps = state => ({
    current: state.current,
    lastWeeks: state.api.lastWeeks
})

const mapDispatchToProps = dispatch => ({
    setNewWeek: (lastWeeks, current) => dispatch(setNewWeek(lastWeeks, current))
})

ChooseWeek.propTypes = {
    setNewWeek: PropTypes.object.isRequired,
    current: PropTypes.number.isRequired,
    lastWeeks: PropTypes.array.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(ChooseWeek)