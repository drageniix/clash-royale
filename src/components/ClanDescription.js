import React from 'react'
import { connect } from 'react-redux'

const ClanDescription = ({admin}) => (
    <section className="description">
        <div className="description__reqs">
            <h2>Clan Requirements</h2>
            <p><strong>Member:</strong> Donate at least {admin.minDonations} cards weekly and participate in {admin.minWarBattles}/10 war battles, missing a maximum of 1 war day battle when you participated in collection day.</p>
            <p><strong>Elder:</strong> Have at least {admin.minTrophies} trophies and donate more than {admin.elderDonations} cards. Must participate in at least {admin.elderWarBattles}/10 wars. Kicking members and sending invitations to those below basic member requirements requires permission.</p>
        </div>
    </section>
)

export default connect(state => ({
    admin: state.admin
}))(ClanDescription)