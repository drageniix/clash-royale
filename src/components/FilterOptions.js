import React from 'react';
import { getSelectedFilterClass } from '../redux/selectors';
import { setFilter } from '../redux/actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class FilterOptions extends React.Component {
    onFilter = filterType => () => {
        this.props.onFilter(filterType);
    };

    render() {
        return (
            <div className="filter">
                <div className="filter__vaporware" />
                <ul className="filter__list">
                    <li
                        className={this.props.getSelectedFilterClass('none')}
                        onClick={this.onFilter('byNone')}
                    >
                        All Members
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'promotion'
                        )}
                        onClick={this.onFilter('byPromotion')}
                    >
                        Promotion
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'probation'
                        )}
                        onClick={this.onFilter('byProbation')}
                    >
                        Probation
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'demotion'
                        )}
                        onClick={this.onFilter('byDemotion')}
                    >
                        Demotion
                    </li>
                </ul>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    getSelectedFilterClass: filter =>
        getSelectedFilterClass(state.filters.filter, filter)
});

const mapDispatchToProps = dispatch => ({
    onFilter: filterType => dispatch(setFilter[filterType]())
});

FilterOptions.propTypes = {
    getSelectedFilterClass: PropTypes.func.isRequired,
    onFilter: PropTypes.func.isRequired
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilterOptions);
