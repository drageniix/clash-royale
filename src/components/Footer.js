import moment from 'moment'
import React from 'react'
import { connect } from 'react-redux'

export const Footer = ({ time, discord }) => (
    <footer className="footer">
        <div className="footer__links">
            <a href={discord}><div className="icon--discord"></div></a>
        </div>
        <p className="footer__updated">Clan data last updated {time && moment.unix(time).fromNow()}.</p>
        <p className="footer__credits">@meliaesc</p>
    </footer>
)

export default connect(state => ({
    time : state.time,
    discord: state.discord
}))(Footer)