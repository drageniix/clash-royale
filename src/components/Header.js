import React from 'react'
import { connect } from 'react-redux'

export const Header = ({ name, discord }) => (
    <header className="header">
        <div class="header__content">
            <img src="./favicon.png" />
            <div className="header__text">
                <h1>{name}</h1>
                <p>2+ Years Strong! Come chat with us on <a href={discord}>Discord</a>.</p>
            </div>
        </div>
    </header>
)

export default connect(state => ({
    name: state.api.clan.name,
    discord: state.api.discord
}))(Header)
