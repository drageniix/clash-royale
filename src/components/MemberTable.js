import React from 'react'


class MemberTable extends React.Component {
    search = (member, event) => {
        this.props.setSearchResult(member.name)
        event.preventDefault()
    }

    render(){
        return  (
            <section>
                <table className="clanMembers">
                    <thead className="clanMembers__header">
                        <tr>
                            <th>Rank</th>
                            <th>Name</th>
                            <th>Donations</th>
                            <th>Wars /10</th>
                            <th>Missed</th>
                        </tr>
                    </thead>
                    {this.props.members.map((member, index) => (
                        <tr onClick={this.search.bind(this, member)} key={index} className={(member.eligibleForPromotion ? "promotion " : member.dangerOfDemotion ? "demotion " : "") + "clanMembers__row"}>
                            <td> #{member.rank}</td>
                            <td> {member.name}</td>
                            <td> {member.donations} ({((member.donations / member.donationsReceived) * 100).toFixed(1)}%)</td>
                            <td> {member.warBattles}</td>
                            <td> {member.missedWarBattles}</td>
                        </tr>
                    ))}
                </table>
            </section>
        )
    }
}

export default MemberTable