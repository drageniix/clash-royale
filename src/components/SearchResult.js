import React from 'react'
import { capitalizeFirstLetter, getMemberColor } from '../utilities/basic'
import { getSearchResult } from '../redux/selectors'
import { connect } from 'react-redux'

const SearchResult = ({ member }) => {
    if (!member) {
        return <section />
    } else {
        return (
            <section className="searchResult" id={getMemberColor(member)}>
                <table className="searchResult__table">
                    <tr>
                        <td>Name</td>
                        <td>{member.name}</td>
                    </tr>
                    <tr>
                        <td>Tag</td>
                        <td>{member.tag}</td>
                    </tr>
                    <tr>
                        <td>Trophies</td>
                        <td>{member.trophies}</td>
                    </tr>
                    <tr>
                        <td>Role</td>
                        <td>{capitalizeFirstLetter(member.role)}</td>
                    </tr>
                    <tr>
                        <td>Donations</td>
                        <td>{member.donations}</td>
                    </tr>
                    <tr>
                        <td>Donations Received</td>
                        <td>{member.donationsReceived}</td>
                    </tr>
                    <tr>
                        <td>War Battles (Last 10)</td>
                        <td>{member.wins}W - {member.warBattles - member.wins}L</td>
                    </tr>
                    <tr>
                        <td>Missed War Battles</td>
                        <td>{member.missedWarBattles}</td>
                    </tr>
                </table>
            </section>
        )
    }
}

export default connect(state => ({
    member: getSearchResult(state.query, state.members)
}))(SearchResult)
