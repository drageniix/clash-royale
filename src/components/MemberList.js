import React from 'react'
import FilterOptions from './FilterOptions';
import SortOptions from './SortOptions';
import { capitalizeFirstLetter, checkDonations } from '../utilities/basic'
import { getMembers } from '../redux/selectors'
import { setQuery } from '../redux/actions'
import { connect } from 'react-redux'

export class MemberList extends React.Component {
    onClick = memberTag => e => {
        window.scroll({ top: 200, left: 0, behavior: 'smooth' }) //can't use in tests
        this.props.onClick(memberTag)
    }

    render() {
        return (
            <section className="filter">
                <FilterOptions />
                <table className="clanMembers">
                    <SortOptions />
                    {this.props.members.map((member, index) => (
                        <tr key={index} className={"clanMembers__" + (member.eligibleForPromotion ? "promotion" : member.onProbation ? "probation" : member.dangerOfDemotion ? "demotion" : "row")}>
                            <td className="align-left"> #{member.clanRank} ({member.trophies})</td>
                            <td className="align-right">{capitalizeFirstLetter(member.role)}</td>
                            <td className="pointer align-left member-link" onClick={this.onClick(member.tag)}>
                                {member.name}
                            </td>
                            <td className="align-right">{checkDonations(member)}</td>
                            <td className="align-right">{member.warBattles}{member.missedWarBattles ? (" (" + member.missedWarBattles + "M)") : ""}</td>
                        </tr>
                    ))}
                </table>
            </section>
        )
    }
}

const mapStateToProps = state => ({
    members: getMembers(state.api.members, state.filters)
})

const mapDispatchToProps = dispatch => ({
    onClick: query => dispatch(setQuery(query))
})

export default connect(mapStateToProps, mapDispatchToProps)(MemberList)