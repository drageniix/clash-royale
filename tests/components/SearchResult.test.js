import React from 'react'
import { SearchResult } from '../../src/components/SearchResult'

test('render: Search Result', () => {
    const wrapper = shallow(
        <SearchResult member={clan.members[0]} />
    )
    expect(wrapper).toMatchSnapshot()
})