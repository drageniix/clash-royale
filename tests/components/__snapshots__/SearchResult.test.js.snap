// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render: Search Result 1`] = `
<section
  className="searchResult"
  id="elder"
>
  <table
    className="searchResult__table"
  >
    <tr>
      <td>
        Name
      </td>
      <td>
        Goofyjay2
      </td>
    </tr>
    <tr>
      <td>
        Tag
      </td>
      <td>
        #PYP8VCRG
      </td>
    </tr>
    <tr>
      <td>
        Trophies
      </td>
      <td>
        4704
      </td>
    </tr>
    <tr>
      <td>
        Role
      </td>
      <td>
        Elder
      </td>
    </tr>
    <tr>
      <td>
        Donations
      </td>
      <td>
        1230
      </td>
    </tr>
    <tr>
      <td>
        Donations Received
      </td>
      <td>
        720
      </td>
    </tr>
    <tr>
      <td>
        War Battles (Last 10)
      </td>
      <td>
        8
        W - 
        3
        L
      </td>
    </tr>
    <tr>
      <td>
        Missed War Battles
      </td>
      <td>
        0
      </td>
    </tr>
  </table>
</section>
`;
